source/core/index.ts

```diff
@@ -1254,7 +1254,7 @@ export default class Request extends Duplex implements RequestEvents<Request> {
 
 	async _makeRequest(): Promise<void> {
 		const {options} = this;
-		const {url, headers, request, agent, timeout} = options;
```

Comment:
Can't we just place the line above after the beforeRequest hooks have been called?

Reply:
I didn't want to make more changes than necessary, but it's ok if `timeout` and `request` should be restored with the `beforeRequest` ones.